daemon/policy.go

```diff
@@ -203,13 +206,19 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 	}
 
 	d.policy.Mutex.Lock()
-	if opts != nil && opts.Replace {
-		for _, r := range rules {
-			tmp := d.policy.SearchRLocked(r.Labels)
-			if len(tmp) > 0 {
-				d.policy.DeleteByLabelsLocked(r.Labels)
+	if opts != nil {
+		if opts.Replace {
+			for _, r := range rules {
+				tmp := d.policy.SearchRLocked(r.Labels)
+				if len(tmp) > 0 {
+					d.policy.DeleteByLabelsLocked(r.Labels)
+				}
```

Comment:
Is this still necessary given that the new option will also take care of deleting any rule this loop might delete?

Reply:
Yes, there are other parts in the code that use the `Replace: true` that is not related with k8s.